### PR TITLE
New Resource: `azurerm_video_indexer_account`

### PR DESCRIPTION
**Pull request by mbfrahry**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR adds `azurerm_video_indexer_account` as a new resource.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* **New Resource:** `azurerm_video_indexer_account` [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @mbfrahry! Could you take a look at the comments I left in-line?

File: internal/services/videoindexer/account_resource.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.VideoIndexer.AccountClient
+			metadata.Logger.Infof("deleting %s", id)
+
+			if _, err := client.Delete(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %v", id, err)
+			}
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply internal/services/videoindexer/account_resource.go:245]: Good shout! That does look weird

File: internal/services/videoindexer/account_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	})
+}
+
+func TestAccVideoIndexerAccount_userAssignedIdentity(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_video_indexer_account", "test")
+	r := AccountResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply internal/services/videoindexer/account_resource_test.go:40]: Ooo yep yep. Ty ty

File: internal/services/videoindexer/account_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retreiving %s: %v", id, err)
+	}
+	if response.WasNotFound(resp.HttpResponse) {
+		return utils.Bool(false), nil
+	}
+	return utils.Bool(true), nil
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply internal/services/videoindexer/account_resource_test.go:101]: Nah, you're right. Copy pasta strikes again

**Review by stephybun — APPROVED** [review]
Thanks @mbfrahry LGTM 👍

**Pull request merged**